### PR TITLE
Create CITATION.bib

### PR DESCRIPTION
**Pull request by tmigot**
Add missing bib file with the citation.

Link to Zenodo notice: https://zenodo.org/record/3929908

### CODE REVIEW

File: CITATION.bib [hunk 1]
```diff
@@ -0,0 +1,8 @@
+@Misc{orban-hsl-2019,
+  author = {D. Orban and {contributors}},
+  title = {{HSL.jl}: A Julia interface to the HSL packages for sparse linear algebra},
```
Comment (dpo) [inline CITATION.bib:3]:
```suggestion
  title = {{HSL.jl}: A Julia interface to the {HSL} Mathematical Software Library},
```

File: CITATION.bib [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  month = {February},
+  howpublished = {\url{https://github.com/JuliaSmoothOptimizers/HSL.jl}},
+  year = {2019},
```
Comment (dpo) [inline CITATION.bib:6]:
```suggestion
  year = {2021},
```

**Review by dpo — CHANGES_REQUESTED** [review]

File: CITATION.bib (hunk 2 above)
Comment (tmigot) [reply CITATION.bib]: Thank you

File: CITATION.bib [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@Misc{orban-hsl-2021,
+  author = {D. Orban and {contributors}},
+  title = {{HSL.jl}: A Julia interface to the {HSL} Mathematical Software Library},
```
Comment (dpo) [inline CITATION.bib:3]:
```suggestion
  title = {{HSL.jl}: A {J}ulia interface to the {HSL} Mathematical Software Library},
```

**Review by dpo — CHANGES_REQUESTED** [review]
Sorry, one more.

**Comment by dpo** [comment]
Thank you

**Pull request merged**